nova/objects/console_auth_token.py

```diff
@@ -67,9 +67,19 @@ def access_url(self):
                 # top-level 'token' query parameter was removed. The 'path'
                 # parameter is supported in older noVNC versions, so it is
                 # backward compatible.
-                qparams = {'path': '?token=%s' % self.token}
-                return '%s?%s' % (self.access_url_base,
-                                  urlparse.urlencode(qparams))
+                # NOTE(mariusleu): If the path parameter is already
+                # configured in self.access_url_base, then we should
+                # just append the token to it.
+                url_parts = list(urlparse.urlparse(self.access_url_base))
+                query_params = {}
+                # Transform query_params into simple key-value dictionary
+                for k, v in urlparse.parse_qs(url_parts[4]).items():
+                    query_params[k] = v[0]
+                path = query_params.get('path') or ''
+                path = '%s?token=%s' % (path, self.token)
+                query_params['path'] = path
+                url_parts[4] = urlparse.urlencode(query_params)
```

Comment:
How about something like:
```python
if "path" in url_parts[4]:
    url_parts[4]["path"] += "?token=%s" % urlparse.quote(self.token)
else:
    url_parts[4]["path"] = "?token=%s" % urlparse.quote(self.token)
```
?

Reply:
It might work for us, but that check only assures that "path" is anywhere in the URL, not a particular query parameter.

Reply:
`url_parts[4]` is just a string, I don't think this example will work, since it has to be parsed into a dict first.

Reply:
(And btw, `params.get('path') or ''` can be shortened to `params.get('path', '')` -- the default fallback is a feature of the `get()` method.)

Reply:
> `url_parts[4]` is just a string, I don't think this example will work, since it has to be parsed into a dict first.

Ah you're right!